### DISCUSSION TITLE
Async listening to MIDI device

### DISCUSSION THREAD

**Discussion by MeestorX** [melanchall/drywetmidi#193] [Q&A] [answered]
Might be a newbie question, but how do I keep an inputDevice listening without blocking code?
The examples show a Console.readKey() function which blocks all other code while it's waiting for new MIDI data on the inputDevice. If I remove the readKey(), then the it stops listening.
I need my code to continue without blocking and listen for new MIDI messages in the background.

Thank you for any advice you can give!

**Comment by melanchall** [comment, marked as answer]
Example is just simple console app. There is nothing hard to adapt it to your case. Just save `InputDevice` to a field of a class and use the device across your program. Nothing blocking here. Without your code and what you want I can't say more.

**Reply by MeestorX** [reply]
Thank you for your reply. My fault. My inexperience with C# is showing.
Had to remove the "using" in the line:
`using (var inputDevice = InputDevice.GetByName("MIDI Device"))`
so that inputDevice would stay instantiated.

Sorry to bother you, and thank you for making this library!

**Reply by melanchall** [reply]
Thanks for using the library. Please don't forget to dispose the device when you're done with it or when the program exists.

**Comment by MeestorX** [comment]
Thank you.
What code do you recommend to use to dispose of the device properly?

**Reply by melanchall** [reply]
`inputDevice.Dispose()`
